src/main/java/com/microsoft/azure/relay/TokenProvider.java

```diff
@@ -50,19 +50,20 @@ public static TokenProvider createSharedAccessSignatureTokenProvider(String keyN
 	 *         generated
 	 */
 	public CompletableFuture<SecurityToken> getTokenAsync(String urlString, Duration validFor) {
-		if (StringUtil.isNullOrEmpty(urlString)) {
-			// TODO: trace
-//            throw RelayEventSource.Log.ArgumentNull(nameof(audience), this);
-		}
-
-		TimeoutHelper.throwIfNegativeArgument(validFor, "validFor");
 		CompletableFuture<SecurityToken> future = new CompletableFuture<SecurityToken>();
-		try {
-			urlString = normalizeAudience(urlString);
-			future = this.onGetTokenAsync(urlString, validFor);
-		}
-		catch (URISyntaxException e) {
+		if (StringUtil.isNullOrEmpty(urlString)) {
+			IllegalArgumentException e = new IllegalArgumentException("Null url provided for security token");
+			RelayLogger.handledExceptionAsWarning(e, this);
 			future.completeExceptionally(e);
+		} else {
+			try {
+				TimeoutHelper.throwIfNegativeArgument(validFor, "validFor");
```

Comment:
>throwIfNegativeArgument [](start = 18, length = 23)

This parameter validation actually throws and is handled by the catch block. With these changes the parameter validations use two different approaches to this. (Also one traces, the other does not).